release/mkrelease.js

```diff
@@ -94,27 +101,42 @@ async function fetchPRsSinceLastReleaseAndEditReleaseNotes(newRelease, callback)
 {
     var derivedFrom = opt.options.derivedFrom;
     console.log("Derived from %o", derivedFrom);
-    if (derivedFrom !== 'latest')
+
+    try
     {
-        if (!derivedFrom.startsWith('v'))
+        var tag = 'latest';
+        if (derivedFrom !== 'latest')
         {
-            derivedFrom = 'v' + derivedFrom;
+            tag = 'v' + derivedFrom;
         }
-        derivedFrom = 'tags/' + derivedFrom;
-    }
+        var releaseInfo = await octokit.repos.getReleaseByTag({
+            owner,
+            repo,
+            tag
+        });
 
-    var body = await (await httpc.get(GIT_HUB_API_URL_ROOT + "/releases/" + derivedFrom)).readBody();
-    body = JSON.parse(body);
-    if (body.published_at === undefined)
+        var lastReleaseDate = releaseInfo.data.published_at;
+        console.log(`Fetching PRs merged since ${lastReleaseDate}`);
+        try
+        {
+            var results = await octokit.search.issuesAndPullRequests({
+                q:`type:pr+is:merged+repo:${owner}/${repo}+merged:>=${lastReleaseDate}`,
+                order: 'asc',
+                sort: 'created'
+            })
```

Comment:
Not a huge deal if this is hard, but is there a way to filter on branch? This only matters if PRs are done on non-master branches - that _could_ happen if we need to do something like correct release notes or if people are working on a feature together not on master. Idk if we'll ever actually do either of those

Reply:
good point,  let me check